addons/fuse-proxy/pkg/server/server.go

```diff
@@ -0,0 +1,222 @@
+	if err != nil {
+		return 0, fmt.Errorf("failed to get socket file: %v", err)
+	}
+	defer sock1File.Close()
+	// Prepare nsenter command to run fusermount in container's namespace
+	nsenterArgs := []string{
+		"--mount=" + nsPath,
+		s.fusermountPath,
+	}
+	nsenterArgs = append(nsenterArgs, req.Args...)
+
+	cmd := exec.Command("nsenter", nsenterArgs...)
+
+	// Pass the socket FD to fusermount
+	cmd.ExtraFiles = []*os.File{sock1File}
+	cmd.Env = append(os.Environ(),
+		fmt.Sprintf("%s=3", common.EnvFuseCommFD), // FD 3 is the first extra file
+	)
+	// Capture command output for error reporting
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return 0, fmt.Errorf("fusermount failed: %v, output: %s", err, string(output))
+	}
+	// For mount operations, we need to transfer the fd back to the caller
+	if !strings.Contains(strings.Join(req.Args, " "), "-u") {
+		fd, _, err := mfcputil.RecvMsg(socks[0])
+		if err != nil {
+			return 0, fmt.Errorf("failed to receive FD: %v", err)
+		}
+		return fd, nil
+	}
+	return 0, nil
+}
+
+// ensureFuseDevice ensures that /dev/fuse exists in the container's namespace
+func (s *Server) ensureFuseDevice(nspath string) error {
+	// First check if /dev/fuse exists in the container's namespace
+	checkCmd := exec.Command("nsenter", "--mount="+nspath, "stat", "/dev/fuse")
+	if err := checkCmd.Run(); err == nil {
+		// Device exists, no need to create
+		return nil
+	}
+
+	// Device doesn't exist, create it
+	// Note: We need to create the device with mknod in the container's namespace
+	createCmd := exec.Command("nsenter",
+		"--mount="+nspath,
+		"mknod",
+		"/dev/fuse",
+		"c",
+		fmt.Sprintf("%d", FuseMajor),
+		fmt.Sprintf("%d", FuseMinor),
+	)
+
+	if output, err := createCmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("failed to create /dev/fuse: %v, output: %s", err, string(output))
+	}
+
+	// Set permissions to allow user access
+	chmodCmd := exec.Command("nsenter",
+		"--mount="+nspath,
+		"chmod",
+		"666",
+		"/dev/fuse",
+	)
+
+	if output, err := chmodCmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("failed to set permissions on /dev/fuse: %v, output: %s", err, string(output))
+	}
+
+	return nil
+}
+
+// createSocketPair creates a pair of connected Unix domain sockets
+func createSocketPair() ([2]net.Conn, error) {
+	// Create a pair of socket file descriptors
+	fds, err := syscall.Socketpair(syscall.AF_UNIX, syscall.SOCK_STREAM, 0)
+	if err != nil {
+		return [2]net.Conn{}, fmt.Errorf("socketpair failed: %v", err)
+	}
+	// Convert the file descriptors to net.Conn
+	conn1, err1 := net.FileConn(os.NewFile(uintptr(fds[0]), ""))
+	if err1 != nil {
+		return [2]net.Conn{}, fmt.Errorf("failed to create first connection: %v", err1)
+	}
+	conn2, err2 := net.FileConn(os.NewFile(uintptr(fds[1]), ""))
+	if err2 != nil {
+		conn1.Close()
+		return [2]net.Conn{}, fmt.Errorf("failed to create second connection: %v", err2)
+	}
+	return [2]net.Conn{conn1, conn2}, nil
+}
```

Comment:
```suggestion
func createSocketPair() (conns [2]net.Conn, err error) {
	defer func() {
		if err != nil {
			for _, conn := range conns {
				conn.Close()
			}
		}
	}()
	// Create a pair of socket file descriptors
	fds, err := syscall.Socketpair(syscall.AF_UNIX, syscall.SOCK_STREAM, 0)
	if err != nil {
		return [2]net.Conn{}, fmt.Errorf("socketpair failed: %w", err)
	}
	for _, fd := range fds {
		conn, err := net.FileConn(os.NewFile(uintptr(fd), ""))
		if err != nil {
			return [2]net.Conn{}, fmt.Errorf("failed to create connection: %w", err)
		}
		conns.append(conn)
	}
	
	return conns, nil
}
```
take it or leave it, the suggestion didn't turn out to be any more concise so I'm not attached to the suggestion

Reply:
looks good!